### ISSUE TITLE
DPP broadcast unsupported for gfx10 present in Batch Norm reduction

### ISSUE THREAD

**Issue by daniellowell** [ROCm/MIOpen#485]
gfx10 does not support broadcast as written with DPP instructions. Need to check architecture and fallback to LDS based reduction until rewrite is complete.

```
<inline asm>:14:20: error: not a valid operand.
v_add_f32 v3 v3 v3 row_bcast:15 row_mask:0xa
^
error: cannot compile inline asm
<inline asm>:15:20: error: not a valid operand.
v_add_f32 v5 v5 v5 row_bcast:15 row_mask:0xa
^
error: cannot compile inline asm
<inline asm>:17:20: error: not a valid operand.
v_add_f32 v3 v3 v3 row_bcast:31 row_mask:0xc
^
error: cannot compile inline asm
<inline asm>:18:20: error: not a valid operand.
v_add_f32 v5 v5 v5 row_bcast:31 row_mask:0xc
^
error: cannot compile inline asm
4 errors generated.
```

**Comment by daniellowell** [comment]
#493 Works around this issue, so it is no longer a bug, but we need a design of this section of the code for gfx10.

**Comment by muralinr** [comment]
All openCL kernels use __AMDGCN__ flag. This flag comes during kernel invocation.  Batchnorm code uses this __AMDGCN__ flag to use inline ASM code.  May be we should disable this flag when "-mcpu" is undefined.

#ifndef MIOPEN_USE_AMDGCN
#if defined(__AMDGCN__) && !(defined(MIO_BN_GFX103X) && MIO_BN_GFX103X)
#define MIOPEN_USE_AMDGCN 1
#else
#define MIOPEN_USE_AMDGCN 0
#endif
#endif

**Comment by ppanchad-amd** [comment]
Ticket no longer relevant. Closing ticket.
